radix-engine/src/model/resource_manager.rs

```diff
@@ -63,30 +68,21 @@ impl MethodEntry {
         &self.auth
     }
 
-    pub fn get_update_auth(&self, args: &[ScryptoValue]) -> &MethodAuthorization {
-        let method: String = match scrypto_decode(&args[0].raw) {
-            Ok(m) => m,
-            _ => return &MethodAuthorization::Unsupported,
-        };
-        match method.as_str() {
-            "lock" | "update" => &self.update_auth,
-            _ => &MethodAuthorization::Unsupported,
+    pub fn get_update_auth(&self, method: MethodAccessRuleMethod) -> &MethodAuthorization {
```

Comment:
`MethodAccessRuleMethod`  doesn't seem to be needed.

Reply:
This is there to maintain what the interface would like once we do separate out MethodAccessRule, ie a rule per method

Reply:
Got it!